continuous_integration/environment-3.10-dev.yaml

```diff
@@ -3,6 +3,7 @@ channels:
 - conda-forge
 - nodefaults
 dependencies:
+- c-compiler
```

Comment:
Despite this being tightly pinned in conda builds, I don't think there's an easy way to pin gcc in the conda environments? `c-compiler` brings in different compiler packages based on platform, and without the use of conditional selectors I don't think there's a way to pin all of those compiler packages in a single conda environment without causing conflict on at least one platform.

cc @jakirkham if you know of a way to achieve this

Reply:
Could have different environment files per platform

Reply:
Yeah that makes sense - I think that for the sake of minimizing environment files to manage, I'm going to leave things as is since CI seems to be passing with GCC left unpinned. If we do run into failures at some point due to leaving it this way, I'd imagine that platform-specific environment files is the best workaround.

Reply:
Yeah that makes sense. Would probably do the same thing

Happy to revisit if it becomes more relevant